line_profiler/_line_profiler.pyx

```diff
@@ -208,17 +211,23 @@ cdef class LineProfiler:
                 func = func.__func__
             code = func.__code__
         except AttributeError:
-            import warnings
-            warnings.warn("Could not extract a code object for the object %r" % (func,))
-            return
+            try:
+                code = func.__func__.__code__
```

Comment:
That would probably be better for clarity, but I believe that both work, as iirc classmethod is a subclass of types.MethodType and therefore anything else that inherits from types.MethodType will also be an instance of classmethod.

Reply:
Now that I think about it, it might be better to avoid importing `types.MethodType` into Cython code, since iirc Cython would bundle the entire types module if we did that, bloating the binary. I'm just going to remove the isinstance, since the nested try/except should handle any cases that triggered it before.

Reply:
That sounds right, however by removing the isinstance should allow both static methods and classmethods to work. With plain try/excepts, it tries `func.__code__` first, which means static methods will work because the exception is never triggered. For classmethods/bound methods, it'll raise the exception and use `func.__func__.__code__` instead, which works.